src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -4830,41 +4833,56 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
-  <data name="ERR_RefReturnCall2" xml:space="preserve">
-    <value>Cannot return by reference a member of result of '{0}' because the argument passed to parameter '{1}' cannot be returned by reference</value>
+    <value>Fields of static readonly field '{0}' cannot be returned by writable reference</value>
   </data>
   <data name="ERR_RefReturnParameter" xml:space="preserve">
     <value>Cannot return a parameter by reference '{0}' because it is not a ref or out parameter</value>
   </data>
   <data name="ERR_RefReturnParameter2" xml:space="preserve">
-    <value>Cannot return by reference a member of parameter '{0}' by reference because it is not a ref or out parameter</value>
+    <value>Cannot return by reference a member of parameter '{0}' because it is not a ref or out parameter</value>
   </data>
   <data name="ERR_RefReturnLocal" xml:space="preserve">
     <value>Cannot return local '{0}' by reference because it is not a ref local</value>
   </data>
   <data name="ERR_RefReturnLocal2" xml:space="preserve">
     <value>Cannot return a member of local '{0}' by reference because it is not a ref local</value>
   </data>
-  <data name="ERR_RefReturnReceiver" xml:space="preserve">
-    <value>Cannot return '{0}' by reference because its receiver may not be returned by reference</value>
-  </data>
   <data name="ERR_RefReturnStructThis" xml:space="preserve">
     <value>Struct members cannot return 'this' or other instance members by reference</value>
   </data>
+  <data name="ERR_EscapeOther" xml:space="preserve">
+    <value>Expression cannot be used in this context because it may indirectly expose variables outside of their declaration scope</value>
+  </data>
+  <data name="ERR_EscapeLocal" xml:space="preserve">
+    <value>Cannot use local '{0}' in this context because it may expose referenced variables outside of their declaration scope</value>
+  </data>
+  <data name="ERR_EscapeCall" xml:space="preserve">
+    <value>Cannot use a result of '{0}' in this context because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="ERR_EscapeCall2" xml:space="preserve">
+    <value>Cannot use a member of result of '{0}' in this context because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="ERR_CallArgMixing" xml:space="preserve">
+    <value>This combination of arguments to '{0}' is disallowed because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="ERR_MismatchedRefEscapeInTernary" xml:space="preserve">
+    <value>Branches of a ref ternary operator cannot refer to variables with incompatible declaration scopes</value>
+  </data>
+  <data name="ERR_MismatchedRefEscapeInTernary" xml:space="preserve">
+    <value>Branches of a ref ternary operator cannot refer to variables with incompatible declaration scopes</value>
+  </data>
```

Comment:
Duplicated entry?

Reply:
Surprising that this is not an error.

---
In reply to: [135930406](https://github.com/dotnet/roslyn/pull/21334#discussion_r135930406) [](ancestors = 135930406)